### DISCUSSION TITLE
Signed Windows Installer

### DISCUSSION THREAD

**Discussion by dlemstra** [ImageMagick/ImageMagick6#284] [Announcements]
Today our code signing certificate will expire. For many years [LeaderSSL](https://www.leaderssl.nl/) sponsored us with a code singing certificate but they are no longer able to do so. Since June of 2023 the CA/B Forum requires that OV code signing private keys be stored on a FIPS 140-2 Level 2 or Common Criteria Level EAL4+ certified device. This means we are no longer able to export our code signing certificate with its private key and use this in GitHub actions. We would now either need to have our own GitHub agent and hardware token or use a cloud solution (e.g. [digicert](https://docs.digicert.com/en/software-trust-manager/ci-cd-integrations/plugins/github-custom-action-for-keypair-signing.html)). Our preference would be to use a cloud solution that integrates with GitHub. Digicert seems to be our only option now but a certificate there would cost [$629](https://order.digicert.com/step1/code_signing) (tax excluded) for a single year. If your organization requires a signed installer then please consider sponsoring us with a code signing certificate. Please reach out to @dlemstra for questions or in case of a sponsorship.

**Comment by urban-warrior** [comment]
@dlemstra says "Thanks for all the options that were offered. We decided to use Azure Code Signing and this means we are now able to sign our binaries again. You can read here how easy it was to set this up here: https://github.com/dlemstra/github-stories/tree/main/2023/ImageMagick%20now%20uses%20Azure%20Code%20Signing."
